### PR TITLE
Feature/osdi model

### PR DESCRIPTION
**Pull request by UnderpantsGnome**
Implement the OSDI data model for AdvocacyCampaign and Event #22 

Replace Features with Specs #29

### CODE REVIEW

File: spec/support/request_helpers.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+module Requests
+  module JsonHelpers
```
Comment (bnd5k) [inline spec/support/request_helpers.rb]: I love this.  Super helpful. :)

**Comment by AaronSofaer** [comment]
If there are indentation concerns or other style nits we should just enforce rubocop's rules, since they're more or less the canon of Ruby style.

File: app/controllers/v1/events_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        redirect_to(action: :show, id: existing_event.id)
+      else
+        super
+      end
+
+    end
+
+    private
+
+      def event_lookup_params
```
Comment (UnderpantsGnome) [inline app/controllers/v1/events_controller.rb]: I've seen it go both ways, I honestly don't know that there's a one true rule on this. I can bump them to the left if that's a norm or preference for the codebases.

My rationale is that the extra indent helps me know at a glance that I'm in a private/protected section of a file.

Comment (bnd5k) [inline app/controllers/v1/events_controller.rb]: Ahh, gotcha.  I wasn't sure if it was intentional.   Keep it.

**Pull request merged**